examples/web-wagmi/src/publications/UseCollect.tsx

```diff
@@ -22,9 +25,33 @@ type CollectButtonProps = {
 
 function CollectButton({ collector, publication }: CollectButtonProps) {
   const { execute: collect, error, isPending } = useCollect({ collector, publication });
+  const {
+    execute: selfFundedCollect,
+    isPending: isSelfFundedCollectFallbackPending,
+    error: selfFundedError,
+  } = useSelfFundedFallback();
+  const [hasRetriedWithSelfFunded, setHasRetriedWithSelfFunded] = useState(false);
 
   const isCollected = publication.hasCollectedByMe;
 
+  const handleCollect = async () => {
+    const attempt = await collect();
+    if (attempt.isSuccess()) {
+      return;
+    }
+    if (supportsSelfFundedFallback(attempt.error)) {
+      const retry = window.confirm(
+        'We cannot cover the transaction costs at this time. Do you want to retry with your own MATIC?',
+      );
+
+      setHasRetriedWithSelfFunded(true);
+
+      if (retry) {
+        await selfFundedCollect(attempt.error.fallback);
```

Comment:
Do you think we can move this to use the self-contained hook approach?

Reply:
of course, done 👍